pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kernelconfig/component.go

```diff
@@ -33,7 +38,21 @@ func (component) Name() string {
 	return "kernel-config"
 }
 
-func (component) Config(_ components.Context) ([]extensionsv1alpha1.Unit, []extensionsv1alpha1.File, error) {
+func (component) Config(ctx components.Context) ([]extensionsv1alpha1.Unit, []extensionsv1alpha1.File, error) {
+	var newData = data
+	if (ctx.KubeletConfigParameters.ProtectKernelDefaults != nil && *ctx.KubeletConfigParameters.ProtectKernelDefaults) ||
+		(ctx.KubeletConfigParameters.ProtectKernelDefaults == nil && ctx.KubernetesVersion != nil && version.ConstraintK8sGreaterEqual126.Check(ctx.KubernetesVersion)) {
```

Comment:
I find it not ideal that we have to make this check two times (see [here](https://github.com/gardener/gardener/blob/c38a317bb0dccea0f25546acdffc39e697d0588a/pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kubelet/config.go#L237)) but I guess this is out of scope of this PR since this can also concern other configurations. Can we also add a comment on this check to clarify why it is made.

Reply:
Can you add a helper function for this which can be used [here](https://github.com/gardener/gardener/blob/c38a317bb0dccea0f25546acdffc39e697d0588a/pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kubelet/config.go#L237) as well?

```
func ProtectedKernelDefaultsEnabled(kubeletConfigParameters, k8sVersion) bool
```